### PR TITLE
Update docs and CI references for `v0.38.x` backport branch

### PR DESCRIPTION
**Pull request by thanethomson**
Partially addresses #33 (specifically #595), following the backport branch creation steps in https://github.com/cometbft/cometbft/blob/main/RELEASES.md#creating-a-backport-branch

Can be reviewed commit by commit.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by jmalicevic — APPROVED** [review]

File: .github/workflows/fuzz-nightly.yml [hunk 1]
```diff
@@ -5,7 +5,8 @@ on:
   schedule:
     - cron: '0 3 * * *'
   pull_request:
-    branches: [main]
+    branches:
+      - v0.38.x
```
[the remark this answers is not in the document]
Comment (thanethomson) [reply .github/workflows/fuzz-nightly.yml:9]: No, YAML allows one to represent arrays both ways. We tend to use this convention more than the `[main]` one though, so I thought I'd adapt it to the conventional way.

File: types/block.go [hunk 2]
```diff
@@ -1010,9 +1010,7 @@ func CommitFromProto(cp *cmtproto.Commit) (*Commit, error) {
 		return nil, errors.New("nil Commit")
 	}
 
-	var (
-		commit = new(Commit)
-	)
+	commit := new(Commit)
```
Comment (lasarojc) [inline types/block.go:1013]: not needed. do it in main.

File: spec/p2p/v0.34/README.md [hunk 3]
```diff
@@ -52,18 +52,18 @@ which implements the following abstractions:
 
 Existing documentation referring to the p2p layer:
 
-- <https://github.com/cometbft/cometbft/tree/main/spec/p2p>: p2p-related
+- <https://github.com/cometbft/cometbft/tree/v0.38.x/spec/p2p>: p2p-related
```
Comment (lasarojc) [inline spec/p2p/v0.34/README.md:55]: should we make these links relative, so they get locked by branch automatically?

**Review by lasarojc — APPROVED** [review]
Some changes are not needed and should be done on main instead.
Some of the links replaced could be made relative, on main, so there is no need to update them in future forks.
Other than that, LGTM

File: spec/p2p/v0.34/README.md (hunk 3 above)
Comment (thanethomson) [reply spec/p2p/v0.34/README.md:55]: I think we need a bigger effort to make as many of the docs' links relative as we can.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -1,13 +1,12 @@
 name: Build
 # Tests runs different tests (test_abci_apps, test_abci_cli, test_apps)
-# This workflow runs on every push to main or release branch and  every pull requests
+# This workflow runs on every push to v0.38.x and every pull request
```
[the remark this answers is not in the document]
Comment (thanethomson) [reply .github/workflows/build.yml:3]: Will do that in a follow-up PR :+1:

**Pull request merged**